Lib/tempfile.py

```diff
@@ -521,17 +533,23 @@ def __iter__(self):
 
 def NamedTemporaryFile(mode='w+b', buffering=-1, encoding=None,
                        newline=None, suffix=None, prefix=None,
-                       dir=None, delete=True, *, errors=None):
+                       dir=None, delete=True, *, errors=None,
+                       delete_on_close=True):
     """Create and return a temporary file.
     Arguments:
     'prefix', 'suffix', 'dir' -- as for mkstemp.
     'mode' -- the mode argument to io.open (default "w+b").
     'buffering' -- the buffer size argument to io.open (default -1).
     'encoding' -- the encoding argument to io.open (default None)
     'newline' -- the newline argument to io.open (default None)
-    'delete' -- whether the file is deleted on close (default True).
+    'delete' -- whether the file is deleted at the end.  When exactly file gets
+       deleted (either on close or on context manager exit) is determined by
+       parameter delete_on_close.  (default True).
     'errors' -- the errors argument to io.open (default None)
     The file is created as mkstemp() would do it.
+    'delete_on_close' -- if 'delete', then determines whether file gets deleted
+       on close. Otherwise it gets deleted on context manager exit
+       (default True)
```

Comment:
The explanation of this parameter shouldn't come after "the file is created as mkstemp() would do it". I think it's better if it immediately follows the `delete` parameter.  
```suggestion
```